### PR TITLE
RP-548 fix  fix consolidated configuration inconsistencies

### PR DESCRIPTION
**Pull request by ftdholbrook**

### CODE REVIEW

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java [hunk 1]
```diff
@@ -202,8 +210,27 @@ public DefaultConfigpropsService(final ConfigpropsConsolidationService configpro
      */
     private <T> Optional<T> condenseConfigurations(final Map<String, T> configurations, final String configTypeKey) {
         HashSet<T> configurationSet = Sets.newHashSet(configurations.values());
-        //one or none are expected
+        //rules for those with more than one configuration expected
         if (configurationSet.size() > 1) {
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java]: this is the core of the change, the rules are somewhat of an educated guess on my part, please comment on RP-548 if they need some change.

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java [hunk 2]
```diff
@@ -53,24 +66,25 @@ public DefaultConfigpropsService(final ConfigpropsConsolidationService configpro
 
     @Override
     public Optional<ReportingSystemSettings> defaultReportingSystemProperties() {
+        final String configTypeKey = "reporting" ;
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java]: was this meant to be extracted?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java [hunk 3]
```diff
@@ -202,8 +210,27 @@ public DefaultConfigpropsService(final ConfigpropsConsolidationService configpro
...
+        //rules for those with more than one configuration expected
         if (configurationSet.size() > 1) {
+            switch (configTypeKey) {
+                case ArchiveConfigTypeKey:
+                    if (configurations.containsKey(RDW_REPORTING_REPORT_PROCESSOR.getName())) {
+                        return Optional.of(configurations.get(RDW_REPORTING_REPORT_PROCESSOR.getName()));
+                    }
+                    break;
+                case DatasourcesReportingRoConfigTypeKey:
+                case DatasourcesReportingRwConfigTypeKey:
```
Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java]: reporting_rw is more of an ingest thing. I don't think reporting-service even uses rw access.

Comment (esotrope) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java]: what about for user created reports and groups?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java (hunk 2 above)
Comment (ftdholbrook) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java]: reporting is kind of a special case, so I didn't but I guess I can to be consistent.

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java (hunk 3 above)
Comment (ftdholbrook) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java]: @esotrope  this is picking an "exemplar" if there is more than one config for some reason, so it is basically "fall back to this one if we can't consolidate it down to one". So on some level, one is as good as another as long as it is correct.

Comment (esotrope) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java]: i'll trust you on this one : j

**Pull request merged**